src/ir_passes/mono/monodeferred.rs

```diff
@@ -227,11 +230,17 @@ impl<'a, 'pass: 'a> MonoDeferred<'a, 'pass> {
         while i < bound {
             let index = Underlying::new(*index);
             self.monosig.binding.insert(index, i);
+            let mut nlets: u32 = 0; // count p_lets as they generate assignments we have to pop later.
             for cmd in body.iter() {
+                nlets += matches!(&cmd, ir::Command::Let(_)) as u32;
                 let cmd = self.command(cmd);
                 self.monosig.base.cmds.extend(cmd);
             }
-            self.monosig.binding.pop();
+            for _ in 0..nlets {
```

Comment:
This should probably use [`truncate`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.truncate) or something instead of using a loop to repeatedly pop.